pycqed/instrument_drivers/physical_instruments/_QCC/QCCMicrocode.py

```diff
@@ -0,0 +1,284 @@
+import logging
+import sys
+
+
+def is_number(s):
+    try:
+        float(s)
+        return True
+    except ValueError:
+        return False
+
+
+def get_bin(x, n, Unsigned=True):
+    '''
+    Return the 2's complement of the integer number $x$
+    for a given bitwidth $n$.
+    '''
+    if (not is_number(x)):
+        raise ValueError('get_bin: parameter {} is not a number.'.format(x))
+
+    if Unsigned is True:
+        return '{0:{fill}{width}b}'.format(int(x), fill='0', width=n)
+    else:
+        return '{0:{fill}{width}b}'.format((int(x) + 2**n) % 2**n,
+                                           fill='0', width=n)
+
+
+def bin_to_hex(bin_val, hex_width):
+    if (not is_number(bin_val)):
+        raise ValueError('bin_to_hex: parameter {} is not a number'.format(
+            bin_val))
+
+    return format(int(bin_val, 2), 'X').zfill(hex_width)
+
+
+def check_int(d, left_bound, right_bound):
+    if not isinstance(d, int):
+        raise ValueError("The input parameter {} is not an integer.".format(d))
+
+    if d < left_bound or d > right_bound:
+        raise ValueError("Parameter {} out of range: [{}, {}].".format(
+            d, left_bound, right_bound))
+
+    return True
+
+
+def ToFourBytes(int_data):
+    if not isinstance(int_data, int):
+        raise ValueError(
+            "The input data {} is not an integer".format(int_data))
+
+    return int_data.to_bytes(4, byteorder='little', signed=False)
+
+
+class QCCMicrocode():
+    """
+    This is class is used to implement all functionalities required to generate
+    the microcode stored in the control store of CCLight.
+    """
+
+    def __init__(self):
+        self.microcode = [0]*256
+        self.CS_header = "OpTypeLeft  CW_Left  OpTypeRight  CW_Right  Condition"
+        self.CS_format = "{:5d}       {:3d}      {:6d}     {:6d}     {:5d} "
+        self.type_dict = ["None", "Mw", "Flux", "Meas"]
+        print("initialization finished.")
```

Comment:
Having a random print of `initialization finished.` does not have any meaning to a user

Reply:
Previously used in CC-Light. Not removing for now in case it may lead to confusing from users of system